### PR TITLE
JS: add support for error values in an axios client request

### PR DESCRIPTION
**Pull request by erik-krogh**
An alternative solution to https://github.com/github/codeql/pull/5856 

Also required adding support for rejected promise values to API-graphs, so I'll run a decently sized evaluation. 
[Evaluation (internal link)](https://github.com/dsp-testing/erik-krogh-dca/tree/run/apiGraphs-added-rejected-promise/reports).

### CODE REVIEW

**Comment by japroc** [comment]
Hi, @erik-krogh.

As i see, this tracks `response.data` Access but not `response.status`? https://github.com/github/codeql/pull/5876/files?file-filters%5B%5D=.js&file-filters%5B%5D=.qll#diff-41f01ac33daf353955a87a26062a949a701ec7b4ef1868a0b67b50c463552fe2R275
What about `response.headers`?  https://zetcode.com/javascript/axios/

**Comment by erik-krogh** [comment]
> As i see, this PR allows to track `response.data` accesses but not `response.status`? https://github.com/github/codeql/pull/5876/files?file-filters%5B%5D=.js&file-filters%5B%5D=.qll#diff-41f01ac33daf353955a87a26062a949a701ec7b4ef1868a0b67b50c463552fe2R275
> What about `response.headers`? https://zetcode.com/javascript/axios/

You're right.  

I considered `.status`, but that is just is just a number, and taint-queries don't really care about number values (at least the vast majority of queries).   

Good point about `.headers`, I'll add that in.  
I think I'll just remove the `getMember("data")` part, that way the implementation also matches the non-error case.

**Comment by japroc** [comment]
Hi!

I agree with about about `response.status` value. I cant come up with a case where this value is really important to taint. Possibly tainting status value is redundant. I think that status value may be useful to track on some specific queries, where it's better to implement `isAdditionalTaintStep`.

File: javascript/ql/src/semmle/javascript/frameworks/ClientRequests.qll [hunk 1]
```diff
@@ -247,12 +242,10 @@ module ClientRequest {
       method = "request" and
       result = getOptionArgument(0, "data")
       or
-      (method = "post" or method = "put" or method = "put") and
-      (result = getArgument(1) or result = getOptionArgument(2, "data"))
+      method = ["post", "put", "put"] and
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/frameworks/ClientRequests.qll:245]: Huh, `put` was mentioned twice, but I think we might as well remove the second mention now that we're at it.

**Review by asgerf — APPROVED** [review]

**Pull request merged**